internal/config/types.go

```diff
@@ -480,29 +482,38 @@ func (c *Config) IsCommandServerProxyConfigured() bool {
 }
 
 // isAllowedDir checks if the given path is in the list of allowed directories.
-// It returns true if the path is allowed, false otherwise.
-// If the path is allowed but does not exist, it also logs a warning.
-// It also checks if the path is a file, in which case it checks the parent directory of the file.
+// It works on files and directories.
+// If the path is a file, it checks the directory of the file.
+// If the path is a directory, it checks the directory itself.
+// It also allows subdirectories within the allowed directories.
 func isAllowedDir(path string, allowedDirs []string) (bool, error) {
 	if len(allowedDirs) == 0 {
 		return false, errors.New("no allowed directories configured")
 	}
 
-	directoryPath := path
-	// Check if the path is a file, regex matches when end of string is /<filename>.<extension>
-	isFilePath, err := regexp.MatchString(`/(\w+)\.(\w+)$`, directoryPath)
-	if err != nil {
-		return false, errors.New("error matching path" + directoryPath)
+	// Check if the path is absolute
+	if !filepath.IsAbs(path) {
+		return false, fmt.Errorf("PathError: %s is not absolute", path)
 	}
 
-	if isFilePath {
-		directoryPath = filepath.Dir(directoryPath)
+	// Does the path exist?
+	info, err := os.Stat(path)
+	if err == nil && !info.IsDir() {
+		// Path exists and is not a directory
+		// It's a file so get the parent directory
+		path = filepath.Dir(path)
+	}
```

Comment:
Why do we think this stat and parent is useful?

If this function is called with `isAllowedDir("/etc/nginx/non-existant-file.conf", ["/etc/nginx"])` then it will skip this branch and needs to work.

If this function is called with `isAllowedDir("/etc/nginx/exists.conf", ["/etc/nginx"])` then it will be converted to "/etc/nginx" and it needs to work.  

What are we gaining on this `Stat` and `IsDir` check? The only case I can think of is to try and catch a case where the file does exist and this process doesn't have read permission to it.  But that's not the purpose of this function, and surfacing the error here feels really odd...